slides/aml-12-022618-model-evaluation/index.html

```diff
@@ -0,0 +1,695 @@
+`$$ \frac{1}{c(c-1)}\sum_{j=1}^{c}\sum_{k \neq j}^{c} AUC(j,k)$$`
+- Provost &amp; Domingo, 2000, one vs rest
+`$$ \frac{\sum_{j=1}^{c}p(j) AUC(j,{rest}_j)}{c}$$`
+- https://github.com/scikit-learn/scikit-learn/pull/7663
+???
+---
+# Pickling metrics ?
+
+- Accuracy rarely what you want
+- Problems are rarely balanced
+- Find the right criterion for the task
+- OR pick one arbitrarily, but at least think about it
+- Emphasis on recall or precision?
+- Which classes are the important ones?
+
+???
+---
+# Using metrics in cross-validation
+.tiny-code[
+```python
+X = digits.data / 16.   # data is between 0 and 16
+# 9 vs rest
+y = digits.target == 9
+
+# default scoring for classification is accuracy
+scores_default = cross_val_score(SVC(), X, y)
+
+# providing scoring="accuracy" doesn't change the results
+explicit_accuracy =  cross_val_score(SVC(), X, y, scoring="accuracy")
+
+# using ROC AUC
+roc_auc =  cross_val_score(SVC(), X, digits.target == 9, scoring="roc_auc")
+
+print("Default scoring: {}".format(scores_default))
+print("Explicit accuracy scoring: {}".format(explicit_accuracy))
+print("AUC scoring: {}".format(roc_auc))
+```
+```
+Default scoring: [ 0.9  0.9  0.9]
+Explicit accuracy scoring: [ 0.9  0.9  0.9]
+AUC scoring: [ 0.994  0.99   0.996]
+```
+]
+
+Same for GridSearchCV
+<br />
+Will make GridSearchCV.score use your metric!
+
+???
+---
+# Built-in scoring
+- "scoring" can be string or callable.
+- Strings:
+
+.tiny-code[
+```python
+from sklearn.metrics.scorer import SCORERS
+print("\n".join(sorted(SCORERS.keys())))
+```
+
+```
+accuracy                    precision_macro
+adjusted_rand_score         precision_micro
+average_precision           precision_samples
+f1                          precision_weighted
+f1_macro                    r2
+f1_micro                    recall
+f1_samples                  recall_macro
+f1_weighted                 recall_micro
+log_loss                    recall_samples
+mean_absolute_error         recall_weighted
+mean_squared_error          roc_auc
+median_absolute_error
+precision
+```
+]
+
+???
+---
+class:spacious
+# Providing you your own callable
+- Takes estimator, X, y
+- Returns score – higher is better (always!)
+.smaller[```python
+def accuracy_scoring(est, X, y): return (est.predict(X) == y).mean()
+```]
+???
+---
+# You can access the model!
+.tiny-code[```python
+def few_support_vectors(est, X, y):
+    acc = est.score(X, y)
+    frac_sv = len(est.support_) / np.max(est.support_)
+    # I just made this up, don't actually use this
+    return acc / frac_sv```
+```python
+from sklearn.model_selection import GridSearchCV
+
+param_grid = {'C': np.logspace(-3, 2, 6),
+              'gamma': np.logspace(-3, 2, 6) / X_train.shape[0]}
+grid = GridSearchCV(SVC(), param_grid=param_grid, cv=10)
+grid.fit(X_train, y_train)
+print(grid.best_params_)
+print(grid.score(X_test, y_test))
+print(len(grid.best_estimator_.support_))
+```
+```
+{'C': 10.0, 'gamma': 0.074239049740163321}
```

Comment:
Can you please round all numbers to 3 decimal places?

Reply:
Done!